### PR TITLE
Change FT sensors names for compatibility with icub-models >= 2.0.0 and robots-configuration >= 2.5.0

### PR DESCRIPTION
**Pull request by traversaro**
Fix https://github.com/robotology/icub-models-generator/issues/242 .

icub-models 2.0.0 changed the name of the FT sensors in the URDF from being named `<identifier>_ft_sensor` (like `l_arm_ft_sensor`, `l_leg_ft_sensor`, ...) to `<identifier>_ft` (like `l_arm_ft`, `l_leg_ft`, ...). 
However, the yarprobotinterface configuration files continued to refer to the sensors as `<identifier>_ft_sensor`, creating errors for software that was trying to match sensors find in URDF and sensors as exposed by the YARP's multipleanalogsensorsserver device.

This PR changes all the instances of iCub configuration files to `<identifier>_ft`, restoring compatibility with icub-models 2.0.0, robots-configuration releases >= 2.5.0 and ergocub-software >= 0.3.4, see https://github.com/robotology/robots-configuration/pull/562 .

Note that the name of the joint to which the sensor is attached remained `<identifier>_ft_sensor`, both before and after icub-models 2.0.0 release.

### CODE REVIEW

**Comment by traversaro** [comment]
@HosameldinMohamed do you have any opinion on this change?

File: devices/baseEstimatorV1/app/robots/iCubGazeboV2_5/estimators/fbe-analogsens.xml [hunk 1]
```diff
@@ -12,10 +12,10 @@
...
         <param name="robot">icubSim</param>
         <param name="device_period_in_seconds">0.010</param>
-        <param name="joints_list">("neck_pitch", "neck_roll", "neck_yaw", "torso_pitch", "torso_roll", "torso_yaw", "l_shoulder_pitch", "l_shoulder_roll", "l_shoulder_yaw", "l_elbow", "r_shoulder_pitch", "r_shoulder_roll", "r_shoulder_yaw", "r_elbow", "l_hip_pitch", "l_hip_roll", "l_hip_yaw", "l_knee", "l_ankle_pitch", "l_ankle_roll", "r_hip_pitch", "r_hip_roll", "r_hip_yaw", "r_knee", "r_ankle_pitch", "r_ankle_roll", "l_arm_ft_sensor", "r_arm_ft_sensor", "l_leg_ft_sensor", "r_leg_ft_sensor", "l_foot_ft_sensor", "r_foot_ft_sensor")</param>
+        <param name="joints_list">("neck_pitch", "neck_roll", "neck_yaw", "torso_pitch", "torso_roll", "torso_yaw", "l_shoulder_pitch", "l_shoulder_roll", "l_shoulder_yaw", "l_elbow", "r_shoulder_pitch", "r_shoulder_roll", "r_shoulder_yaw", "r_elbow", "l_hip_pitch", "l_hip_roll", "l_hip_yaw", "l_knee", "l_ankle_pitch", "l_ankle_roll", "r_hip_pitch", "r_hip_roll", "r_hip_yaw", "r_knee", "r_ankle_pitch", "r_ankle_roll", "l_arm_ft", "r_arm_ft", "l_leg_ft", "r_leg_ft", "l_foot_ft", "r_foot_ft")</param>
```
Comment (HosameldinMohamed) [inline devices/baseEstimatorV1/app/robots/iCubGazeboV2_5/estimators/fbe-analogsens.xml:15]: I think the joints' names are still `<identifier>_ft_sensor` so probably this should remain as before?

File: devices/baseEstimatorV1/include/baseEstimatorV1.h [hunk 2]
```diff
@@ -547,7 +547,7 @@ namespace yarp {
 
         const double m_sensor_timeout_in_seconds{2.0}; ///< Timeout to check for sensor measurements during dry run initial check
         const size_t m_nr_of_channels_in_YARP_IMU_sensor{12}; ///< Number of channels available in YARP IMU sensor output port
-        const size_t m_nr_of_channels_in_YARP_FT_sensor{6}; ///< Number of channels available in YARP FT sensor output port
+        const size_t m_nr_of_channels_in_YARP_ft{6}; ///< Number of channels available in YARP FT sensor output port
```
Comment (HosameldinMohamed) [inline devices/baseEstimatorV1/include/baseEstimatorV1.h:550]: here you updated the name of a method in the source code.. I'm not sure it's relevant.. But I guess it's harmless if you update all it's occurrences in the code..

**Review by HosameldinMohamed — COMMENTED** [review]
Seems good except for a couple of places, please see below!

Have you by chance tested this at least in simulated robots?

File: devices/baseEstimatorV1/include/baseEstimatorV1.h (hunk 2 above)
Comment (traversaro) [reply devices/baseEstimatorV1/include/baseEstimatorV1.h:550]: No, you are right. That variable should not be changed.

File: devices/baseEstimatorV1/app/robots/iCubGazeboV2_5/estimators/fbe-analogsens.xml (hunk 1 above)
Comment (traversaro) [reply devices/baseEstimatorV1/app/robots/iCubGazeboV2_5/estimators/fbe-analogsens.xml:15]: Good catch, let me fix this.

**Pull request closed without merging**